File: app/src/main/java/com/cv4j/app/activity/BeautySkinActivity.java
package com.cv4j.app.activity;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.widget.ImageView;

import com.cv4j.app.R;
import com.cv4j.app.app.BaseActivity;
import com.cv4j.core.filters.BeautySkinFilter;
import com.cv4j.rxjava.RxImageData;
import com.safframework.injectview.annotations.InjectExtra;
import com.safframework.injectview.annotations.InjectView;
import com.safframework.injectview.annotations.OnClick;

/**
 * Created by Tony Shen on 2017/4/23.
 */

public class BeautySkinActivity extends BaseActivity {

    @InjectView(R.id.image1)
    ImageView image1;

    @InjectView(R.id.image2)
    ImageView image2;

    @InjectView(R.id.toolbar)
    Toolbar toolbar;

    @InjectExtra(key = "Title")
    String title;

    RxImageData rxImageData;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_beauty_skin);

        initData();
    }

    private void initData() {
        toolbar.setTitle("< "+title);
        Resources res= getResources();
        Bitmap bitmap = BitmapFactory.decodeResource(res, R.drawable.test_beauty_skin2);
        image1.setImageBitmap(bitmap);

        rxImageData = RxImageData.bitmap(bitmap);
        rxImageData.addFilter(new BeautySkinFilter()).into(image2);
    }

    @OnClick(id= R.id.toolbar)
    void clickToolbar() {

        finish();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        rxImageData.recycle();
    }
}

File: app/src/main/java/com/cv4j/app/activity/BeautySkinActivity.kt
package com.cv4j.app.activity

import android.content.res.Resources
import android.graphics.BitmapFactory
import android.os.Bundle
import com.cv4j.app.R
import com.cv4j.app.app.BaseActivity
import com.cv4j.core.filters.face.BeautySkinFilter
import com.cv4j.rxjava.RxImageData
import com.cv4j.rxjava.RxImageData.Companion.bitmap
import kotlinx.android.synthetic.main.activity_beauty_skin.*

/**
 *
 * @FileName:
 *          com.cv4j.app.activity.BeautySkinActivity
 * @author: Tony Shen
 * @date: 2020-05-05 10:57
 * @version: V1.0 <描述当前版本功能>
 */

class BeautySkinActivity : BaseActivity() {

    var title: String? = null

    private lateinit var rxImageData: RxImageData

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_beauty_skin)

        intent.extras?.let {
            title = it.getString("Title","")
        }?:{
            title = ""
        }()

        toolbar.setOnClickListener {
            finish()
        }

        initData()
    }

    private fun initData() {
        toolbar.title = "< $title"
        val res: Resources = resources
        var bitmap = BitmapFactory.decodeResource(res, R.drawable.test_beauty_skin1)
        origin_image1.setImageBitmap(bitmap)
        rxImageData = bitmap(bitmap)
        rxImageData.addFilter(BeautySkinFilter()).into(image1)
        bitmap = BitmapFactory.decodeResource(res, R.drawable.test_beauty_skin2)
        origin_image2.setImageBitmap(bitmap)
        rxImageData = bitmap(bitmap)
        rxImageData.addFilter(BeautySkinFilter()).into(image2)
    }

    override fun onDestroy() {
        super.onDestroy()
        rxImageData.recycle()
    }
}